File: Dhairyashil-Shinde/905-sort-array-by-parity.cpp
class Solution
{
public:
    vector<int> sortArrayByParity(vector<int> &nums)
    {

        int n = nums.size();

        int index = 0;
        int temp;

        for (int i = 0; i < n; i++)
        {
            if (nums[i] % 2 == 0)
            {
                temp = nums[i];
                nums[i] = nums[index];
                nums[index] = temp;
                index++;
            }
        }
        return nums;
    }
};

File: Lokesh_Saana/905-sort-array-by-parity.py
# https://leetcode.com/problems/sort-array-by-parity/
    
class Solution:
    def sortArrayByParity(self, nums: List[int]) -> List[int]:
        index=0
        for i in range(len(nums)):
            if nums[i]%2==0:
                temp=nums[i]
                nums[i]=nums[index]
                nums[index]=temp
                index+=1
        return nums